textattack/shared/attacked_text.py

```diff
@@ -146,6 +147,31 @@ def pos_of_word_index(self, desired_word_idx):
             f"Did not find word from index {desired_word_idx} in flair POS tag"
         )
 
+    def ner_of_word_index(self, desired_word_idx):
+        """Returns the ner tag of the word at index `word_idx`.
+
+        Uses FLAIR ner tagger.
+        """
+        if not self._ner_tags:
+            sentence = Sentence(self.text)
+            textattack.shared.utils.flair_tag(sentence, "ner")
+            self._ner_tags = sentence
+        flair_word_list, flair_ner_list = textattack.shared.utils.zip_flair_result(
+            self._ner_tags, "ner"
+        )
+
+        for word_idx, word in enumerate(flair_word_list):
+            word_idx_in_flair_tags = flair_word_list.index(word)
+            if word_idx == desired_word_idx:
+                return flair_ner_list[word_idx_in_flair_tags]
+            else:
+                flair_word_list = flair_word_list[word_idx_in_flair_tags + 1 :]
+                flair_ner_list = flair_ner_list[word_idx_in_flair_tags + 1 :]
+
+        raise ValueError(
+            f"Did not find word from index {desired_word_idx} in flair POS tag"
+        )
+
```

Comment:
Awesome work! This was the hardest part of everything, I think, and looks like you nailed it!